src/util.js

```diff
@@ -198,6 +198,18 @@ function getNextNonSpaceNonCommentCharacter(text, node) {
   return text.charAt(idx);
 }
 
+function getPreviousNonSpaceCharacter(text, node) {
+  let oldIdx = null;
+  let idx = locStart(node) - 1;
+  const traversalOptions = {backwards: true}
+  while (idx !== oldIdx) {
+    oldIdx = idx;
+    idx = skipSpaces(text, idx, traversalOptions);
+    idx = skipNewline(text, idx, traversalOptions);
+  }
```

Comment:
You don't handle comments. This is tricky to get right, so let's see if we can make it work with getNextNonSpaceCharacter instead